File: solutions/509.Fibonacci_Number/509.Fibonacci_Number.ts
function fib(n: number): number {
    if(n === 0) return 0
    if(n === 1 || n === 2) return 1

    let n1 = 1
    let n2 = 2
    for(let i = 3; i < n; i++) {
        [n1, n2] = [n2, n1 + n2]
    }
    return n2
};

File: solutions/easy/509.Fibonacci_Number/509.Fibonacci_Number.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0
        if n == 1 or n == 2:
            return 1
        n1 = 1
        n2 = 2

        for i in range(3, n):
            n1, n2 = n2, n1 + n2

        return n2
